src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -123,34 +123,80 @@ SyntaxNodeAnalysisContext ctx
 				);
 
 				return;
-			}
-
-			// Literal arguments should always be named
-			foreach( var arg in unnamedArgs ) {
-
-				// Check if the argument type is literal
-				if( arg.Syntax.Expression is LiteralExpressionSyntax ) {
-
+			}
+
+			int numOfSwappableArgs = 0;
+			foreach( var arg in unnamedArgs ) {
+
+				// Literal expressions have their own issue
+				if( arg.Syntax.Expression is LiteralExpressionSyntax ) {
+					// Literal arguments should always be named
 					var fixerContext = CreateFixerContext( ImmutableArray.Create( arg ) );
-
 					ctx.ReportDiagnostic( Diagnostic.Create(
 							descriptor: Diagnostics.LiteralArgShouldBeNamed,
 							location: arg.Syntax.Expression.GetLocation(),
 							properties: fixerContext,
-							messageArgs: arg.ParamName
-						)
-					);
+							messageArgs: arg.ParamName ) );
+					continue;
+				}
+
+				// Take note of the type of argument
+				ITypeSymbol argType = GetArgumentType( ctx, arg.Syntax );
+
+				// Iterate through the other arguments in the list
+				foreach( var compareArg in unnamedArgs ) {
+					if( compareArg.Position == arg.Position ) {
+						continue;
+					}
+
+					// Get the associated parameter of the argument to compare
+					IParameterSymbol param = compareArg.Syntax.DetermineParameter(
+						ctx.SemanticModel,
+						allowParams: false );
+
+					// Check if the argument could apply to the other parameter
+					if( ctx.SemanticModel.Compilation.ClassifyConversion(
+						argType,
+						param.Type ).Exists ) {

```

Comment:
This is kind of hard to read, so pulling it out into a local variable would be good